Stack/Opc.Ua.Core/Security/Certificates/CertificateValidator.cs

```diff
@@ -214,20 +215,40 @@ public virtual async Task Update(SecurityConfiguration configuration)
                 {
                     m_minimumCertificateKeySize = configuration.MinimumCertificateKeySize;
                 }
+                if ((m_protectFlags & ProtectFlags.MinimumECCertificateKeySize) == 0)
```

Comment:
m_semaphore.WaitAsync should be before try block

Reply:
Can't m_semaphore.WaitAsync throw exceptions in certain situations ?